src/local-mode.ts

```diff
@@ -70,6 +75,7 @@ export class LocalModeManager {
 		public docker: Docker,
 		public logger: Logger,
 		public db: Database,
+		private containerId: string | undefined = constants.containerId,
```

Comment:
Just to check, the `| undefined` here is to allow assigning `constants.containerId` right?  Because if it's expected to be able to explicitly pass `undefined` as the value then that will fail due to the default

Reply:
It's supposed to be used either as
```
new LocalModeManager(config, docker, logger, db)
```
from the main code where the manager is created - in such case the ID will be taken from constants (using the environment variable)
or as
```
new LocalModeManager(config, docker, logger, db, 'someSpecificValue')
```
which is used in tests to inject a specific container ID